### PR TITLE
Fix warning when using default stage name and FailOnWarnings is on

### PR DESCRIPTION
**Pull request by GavinZZ**
### Issue #, if available
https://github.com/aws/serverless-application-model/issues/2297

### Description of changes
Before the change, when we turn on `FailOnWarnings` flag and use a default stage name, the deployment will fail because it will generate a warning saying 
```
Warnings found during import: Parse issue: attribute paths. 

Resource $default should start with / 
```

### Description of how you validated changes
Added integ test and ran all unit tests.

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/translator/output/aws-cn/explicit_http_api.json [hunk 1]
```diff
@@ -123,7 +123,7 @@
           },
           "openapi": "3.0.1",
           "paths": {
-            "$default": {
+            "/$default": {
```
Comment (hoffa) [inline tests/translator/output/aws-cn/explicit_http_api.json:126]: This shouldn't change. The template deploys just fine no?

Comment (hoffa) [reply tests/translator/output/aws-cn/explicit_http_api.json:126]: Ignore, just saw the property below.

Comment (GavinZZ) [reply tests/translator/output/aws-cn/explicit_http_api.json:126]: This test has FailedOnWarnings on.

File: samtranslator/model/api/http_api_generator.py [hunk 2]
```diff
@@ -224,6 +225,19 @@ def _add_cors(self) -> None:
...
+    def _update_paths(self):
+        if not self.fail_on_warnings:
+            return
+
+        # Using default stage name generate warning during deployment
+        #   Warnings found during import: Parse issue: attribute paths. 
+        #   Resource $default should start with / (Service: AmazonApiGatewayV2; Status Code: 400; 
+        # Deployment fails when FailOnWarnings is true
+        paths: Optional[Dict[str, Any]] = self.definition_body.get("paths")
+        if DefaultStageName in paths:
```
Comment (hoffa) [inline samtranslator/model/api/http_api_generator.py:237]: If `paths` is `Optional`, will `in` work on `None`?

Comment (hoffa) [reply samtranslator/model/api/http_api_generator.py:237]:
```
>>> 'lol' in None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument of type 'NoneType' is not iterable
```

File: samtranslator/model/api/http_api_generator.py [hunk 3]
```diff
@@ -224,6 +225,19 @@ def _add_cors(self) -> None:
...
         self.definition_body = editor.openapi
 
+    def _update_paths(self):
+        if not self.fail_on_warnings:
+            return
+
+        # Using default stage name generate warning during deployment
+        #   Warnings found during import: Parse issue: attribute paths. 
+        #   Resource $default should start with / (Service: AmazonApiGatewayV2; Status Code: 400; 
+        # Deployment fails when FailOnWarnings is true
```
Comment (hoffa) [inline samtranslator/model/api/http_api_generator.py:235]:
```suggestion
        # Deployment fails when FailOnWarnings is true: https://github.com/aws/serverless-application-model/issues/2297
```

**Review by acristin — APPROVED** [review]

File: samtranslator/model/api/http_api_generator.py [hunk 4]
```diff
@@ -224,6 +225,19 @@ def _add_cors(self) -> None:
         # Assign the OpenApi back to template
         self.definition_body = editor.openapi
 
+    def _update_paths(self):
```
Comment (hoffa) [inline samtranslator/model/api/http_api_generator.py]: More descriptive:

```suggestion
    def _update_default_path(self):
```

**Review by aahung — APPROVED** [review]

**Review by hoffa — APPROVED** [review]

**Pull request merged**